src/cargo/util/interning.rs

```diff
@@ -13,7 +13,19 @@ use std::str;
 use std::sync::Mutex;
 use std::sync::OnceLock;
 
-static STRING_CACHE: OnceLock<Mutex<HashSet<&'static str>>> = OnceLock::new();
+pub static DEFAULT_INTERNED_STRING: InternedString = InternedString { inner: "default" };
+
+fn cache_init() -> std::sync::MutexGuard<'static, HashSet<&'static str>> {
```

Comment:
nit: it would be easier to audit the behavior change if the pulling out of `cache_init` was in its own commit.